### ISSUE TITLE
Distinguish between used RAM and ZFS ARC

### ISSUE THREAD

**Issue by Caian** [cockpit-project/cockpit#19482]
Page: Overview / Metrics

Cockpit will show the RAM as nearly full on systems running ZFS pools due to of ZFS ARC (cache). This doesn't mean that the memory is really full, as ARC is designed to free memory on demand, but makes it harder to identify if the system is really running low on memory.

It would be nice to distinguish between memory used by ARC, maybe show it on the graph as a different color and subtract it from the "used" statistic.

ARC memory is easily accessible through `/proc` at `/proc/spl/kstat/zfs/arcstats`

**Comment by martinpitt** [comment]
This feels like a kernel bug to me, i.e. in the ZFS drivers. That should count into either `Buffers` or `Cached` in /proc/meminfo, instead of getting deducted from `MemoryAvailable`.

**Comment by Caian** [comment]
@martinpitt AFAIK ZFS ARC counts as standard memory, that is allocated by the kernel.

A similar issue with netdata: https://github.com/netdata/netdata/issues/12843

**Comment by martinpitt** [comment]
Right, but that feels odd -- if it's cache and frees itself under memory pressure, it should declare itself as such? We currently don't test/support ZFS, and adding such special cases to  userspace is not something I'm inclined to do, to be honest. I'm keeping this open for now, but we won't jump at this.

**Comment by Caian** [comment]
I agree with you on that. 

Is it possible to modify this behavior in the UI only through a plugin?

**Comment by martinpitt** [comment]
No, I'm afraid not. This is wired pretty [deeply into the bridge](https://github.com/cockpit-project/cockpit/blob/5caea4b59d5bb4654312b23861cec96afbc8de40/src/cockpit/samples.py#L128), which takes it out of /proc/meminfo. Moreover, the "history metrics" aka. [PCP](https://pcp.io/) also implements it that way, i.e. doesn't know about ZFS. That's one reason why we shouldn't just hack this in the bridge, but it should be done in /proc/meminfo.

**Comment by martinpitt** [comment]
As described above, this is a kernel bug from our PoV and we won't hack around it. We don't support/test ZFS.
